gradebookng/impl/src/main/java/org/sakaiproject/grading/impl/GradingServiceImpl.java

```diff
@@ -5035,10 +5036,10 @@ public CommentDefinition getAssignmentScoreComment(String gradebookUid, Long ass
         return commentDefinition;
     }
 
-    public void setAssignmentScoreComment(String gradebookUid, Long assignmentId, String studentUid, String commentText) throws AssessmentNotFoundException {
+    public void setAssignmentScoreComment(String gradebookUid, Long assignmentId, String studentUid, @Nullable String commentText) throws AssessmentNotFoundException {
```

Comment:
@Nullable does not add runtime checks, so it won't protect against an empty or null commentText here.

Reply:
Okay, Comment.commentText can be null, so in theory this is fine. I'm a little worried about what other code is calling comment.getCommentText and expecting it to be non null. Have you checked calls like that?

Reply:
Yes, I have reviewed all the methods that call ```getCommentText()```, but I did not come across any explicit indications where the comment should not be empty.